starknet_devnet/starknet_wrapper.py

```diff
@@ -266,64 +245,49 @@ def get_block_by_number(self, block_number: int):
             raise StarknetDevnetException(message=message)
 
         if block_number >= self.get_number_of_blocks():
-            message = f"Block number too high. There are currently {len(self.__own_blocks)} blocks; got: {block_number}."
+            message = f"Block number too high. There are currently {len(self.__num2block)} blocks; got: {block_number}."
             raise StarknetDevnetException(message=message)
 
-        if block_number in self.__own_blocks:
-            return self.__own_blocks[block_number]
+        if block_number in self.__num2block:
+            return self.__num2block[block_number]
 
         return self.origin.get_block_by_number(block_number)
 
-    async def __store_transaction(self, contract_address: str, status: TxStatus,
-        execution_info: StarknetTransactionExecutionInfo, error_message: str=None, **transaction_details: dict
-    ):
-        new_id = len(self.__transaction_wrappers)
-        hex_new_id = hex(new_id)
-
-        transaction = _generate_transaction_basis(contract_address, status, hex_new_id, **transaction_details)
-        receipt = _generate_transaction_receipt_basis(status, hex_new_id, execution_info)
+    async def __store_transaction(self, transaction_wrapper: TransactionWrapper, error_message):
 
-        if status == TxStatus.REJECTED:
+        if transaction_wrapper.transaction["status"] == TxStatus.REJECTED:
             failure_key = "transaction_failure_reason"
-            transaction[failure_key] = receipt[failure_key] = {
-                "code": StarknetErrorCode.TRANSACTION_FAILED.name,
-                "error_message": error_message,
-                "tx_id": new_id
-            }
+            transaction_wrapper.set_transaction_failure(failure_key,error_message)
         else:
-            await self.__generate_block(transaction, receipt)
+            await self.__generate_block(transaction_wrapper.transaction, transaction_wrapper.receipt)
 
-        self.__transaction_wrappers.append(TransactionWrapper(transaction, receipt))
-        return hex_new_id
+        self.__transaction_wrappers[transaction_wrapper.transaction_hash] = transaction_wrapper
+
+        return transaction_wrapper.transaction_hash
 
     async def __store_deploy_transaction(self, transaction: InternalDeploy, status: TxStatus,
```

Comment:
Can __store_deploy_transaction, __store_invoke_transaction and __store_transaction now be the same function? Because the two first mentioned are very similar (only differing in the wrapper class used - so it could be wrapped outside and passed to the new "merged" function)